vendor/parser-extended/parsetree.mli

```diff
@@ -684,15 +684,21 @@ and class_type_desc =
   | Pcty_extension of extension  (** [%id] *)
   | Pcty_open of open_description * class_type  (** [let open M in CT] *)
 
+and class_sig_self =
+  {
+   pcss_desc:core_type option;
+   pcss_loc : Location.t;
+  }
```

Comment:
Is there really a location if `pcss_desc` is None? I think it's enough to just use `core_type option`, the loc of the type (if any) is the loc of the class_sig_self, as in #2213

Reply:
The loc was used in a few places, I will try to remove these uses